tests/functional/test_snap_dcgm.py

```diff
@@ -1,86 +1,139 @@
-                    f"nc -z localhost {value.lstrip(':')}".split()
-                ), f"{service} is not listening on {value}"
+                    f"nc -z localhost {bind.lstrip(':')}".split()
+                ), f"{service} is not listening on {bind}"
+
+    @classmethod
+    def get_config(cls, config: str) -> str:
+        """Check if a configuration exists in the snap configuration.
+
+        :return: The value of the current configuration
+        """
+        result = subprocess.check_output("sudo snap get dcgm -d".split(), text=True)
+        dcgm_snap_config = json.loads(result.strip())
+        assert config in dcgm_snap_config, f"{config} is not in the snap configuration"
+        return str(dcgm_snap_config[config])
+
+    @classmethod
+    @retry(wait=wait_fixed(2), stop=stop_after_delay(10))
+    def check_metric_config(cls, metric_file: str) -> None:
+        dcgm_exporter_service = "snap.dcgm.dcgm-exporter"
+
+        result = subprocess.check_output(
+            f"sudo systemctl show -p ActiveEnterTimestamp {dcgm_exporter_service}".split(),
+            text=True,
+        )
+
+        start_timestamp = result.strip().split("=")[1]
+
+        result = subprocess.check_output(
+            f"sudo journalctl -u {dcgm_exporter_service} --since '{start_timestamp}'",
+            shell=True,
+            text=True,
+        )
+
+        assert metric_file in result, f"Metric file {metric_file} is not loaded"
+
+    @pytest.mark.parametrize(
+        "service, config, new_value",
+        [
+            ("dcgm.dcgm-exporter", "dcgm-exporter-address", ":9466"),
+            ("dcgm.nv-hostengine", "nv-hostengine-port", "5566"),
+        ],
+    )
+    def test_dcgm_bind_config(self, service: str, config: str, new_value: str) -> None:
+        """Test snap bind configuration."""
+        old_value = self.get_config(config)
+
+        self.set_config(service, config, new_value)
+        self.check_bind_config(service, new_value)
+
+        # Revert back
+        self.set_config(service, config, old_value)
+        self.check_bind_config(service, old_value)
+
+    def test_dcgm_metric_config(self) -> None:
+        service = "dcgm.dcgm-exporter"
+        config = "dcgm-exporter-metrics-file"
+        metric_file = "test-metrics.csv"
+        metric_file_path = os.path.join(os.getenv("SNAP_COMMON"), metric_file)
+
+        self.get_config(config)
+
+        # $SNAP_COMMON requires root permissions to create a file
+        subprocess.check_call(f"sudo touch {metric_file_path}".split())
+
+        self.set_config(service, config, metric_file)
+        self.check_metric_config(metric_file_path)
 
-    # Check new config
-    set_config_and_check(new_value)
+        # Revet back
+        self.set_config(service, config)
+        self.check_metric_config("default-counters.csv")
 
-    # Revert back
-    set_config_and_check(str(old_value))
+        subprocess.check_call(f"sudo rm {metric_file_path}".split())
```

Comment:
nit: is this necessary? The test environment will be destroyed anyway

Reply:
What if you test locally?

Reply:
yeah good point. Although I'm generally terrified of what functional tests may do to my system so I never run them directly on the host :D

Reply:
Yeah it would be nice if we could have functional tests that you could run locally and not worry. For charms it's usually not a problem because the tests happen in a throwaway model in whichever juju controller you configure.

Reply:
Sounds like something interesting to explore

Reply:
(suggestion)
Need to be careful, if something went wrong before this line then it's a side effect. A pytest fixture will be more nice.

Reply:
Actually, I don't need to remove the file explicitly.  As the test operates on the `SNAP_COMMON` of the dcgm, everything, including the folder will be removed during the cleanup by this [line](https://github.com/canonical/dcgm-snap/blob/74e8a9252a6356b8d93319767c4d844510336c29/tests/functional/conftest.py#L22)